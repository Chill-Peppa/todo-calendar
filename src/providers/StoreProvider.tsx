import React, { FC, useState } from 'react';
import { StoreContext } from '../hooks/useStore';

interface IStoreProvider {
  children: React.ReactNode;
}
export const StoreProvider: FC<IStoreProvider> = ({ children }) => {
  const [text, setText] = useState('');

  return (
    <StoreContext.Provider value={{ text, setText }}>
      {children}
    </StoreContext.Provider>
  );
};
