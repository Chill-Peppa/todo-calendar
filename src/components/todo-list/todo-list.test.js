import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { StoreProvider } from '../../providers/StoreProvider';
import TodoList from './todo-list';

const selectedDate = '2022-01-01';

it('should correctly render TodoList', () => {
  const tree = renderer
    .create(
      <StoreProvider>
        <TodoList selectedDate={selectedDate} />
      </StoreProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should renders message if filteredTodoList is empty', () => {
  render(
    <StoreProvider>
      <TodoList selectedDate={selectedDate} />
    </StoreProvider>,
  );
  const message = screen.getByText(
    '–ù–∞ —ç—Ç–æ—Ç –¥–µ–Ω—å —Å–æ–±—ã—Ç–∏–π –Ω–µ –∑–∞–ø–ª–∞–Ω–∏—Ä–æ–≤–∞–Ω–æ. –î–æ–±–∞–≤—å—Ç–µ –≤–∞—à–µ –ø–µ—Ä–≤–æ–µ —Å–æ–±—ã—Ç–∏–µ ü•∫',
  );
  expect(message).toBeInTheDocument();
});
