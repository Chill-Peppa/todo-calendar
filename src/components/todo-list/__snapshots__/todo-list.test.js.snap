// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render TodoList 1`] = `
<div
  className="modal__todo-container"
>
  <span
    className="modal__add-todo"
  >
    –ù–∞ —ç—Ç–æ—Ç –¥–µ–Ω—å —Å–æ–±—ã—Ç–∏–π –Ω–µ –∑–∞–ø–ª–∞–Ω–∏—Ä–æ–≤–∞–Ω–æ. –î–æ–±–∞–≤—å—Ç–µ –≤–∞—à–µ –ø–µ—Ä–≤–æ–µ —Å–æ–±—ã—Ç–∏–µ ü•∫
  </span>
  <div
    className="modal__add-container"
  >
    <input
      className="modal__input"
      data-testid="text"
      id="add-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="–î–æ–±–∞–≤—å—Ç–µ –Ω–æ–≤–æ–µ —Å–æ–±—ã—Ç–∏–µ"
      type="text"
      value=""
    />
    <button
      className="modal__add-button"
      data-testid="add-button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
